src/perf/isolates.md

```diff
@@ -0,0 +1,347 @@
+
+Long-lived isolates are useful when you have a specific process that either
+needs to be run repeatedly throughout the lifetime of your application,
+or if you have a process that runs over a period of time
+and needs to yield multiple return values to the main isolate.
+
+### ReceivePorts and SendPorts
+
+Set up long-lived communication between isolates with two classes
+(in addition to Isolate):
+[`ReceivePort`][] and [`SendPort`][].
+These ports are the only way isolates can communicate with each other.
+
+`Ports` behave similarly to `Streams`,
+in which the `StreamController`
+or `Sink` is created in one isolate,
+and the listener is set up in the other isolate.
+In this analogy,
+the `StreamConroller` is called a `SendPort`,
+and you can "add" messages with the `send()` method.
+`ReceivePort`s are the listeners,
+and when these listeners receive a new message,
+they call a provided callback with the message as an argument.
+
+For an in-depth explanation on setting up two-way
+communication between the main isolate
+and a worker isolate,
+follow the examples in the [Dart documentation][].
+
+[Dart documentation]: {{site.dart-site}}/language/concurrency
+
+## Using platform plugins in isolates
+
+As of Flutter 3.7, you can use platform plugins in background isolates.
+This opens many possibilities to offload heavy,
+platform-dependent computations to an isolate that won't block your UI.
+For example, imagine you're encrypting data
+using a native host API
+(such as an Android API on Android, an iOS API on iOS, and so on).
+Previously, [marshaling data][] to the host platform could waste UI thread time,
+and can now be done in a background isolate.
+
+Platform channel isolates use the [`BackgroundIsolateBinaryMessenger`][] API.
+The following snippet shows an example of using 
+the shared_preferences package in a background isolate.
+
+<?code-excerpt "lib/isolate_binary_messenger.dart"?>
+```dart
+import 'dart:isolate';
+
+import 'package:flutter/services.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+void main() {
+  // Identify the root isolate to pass to the background isolate.
+  RootIsolateToken rootIsolateToken = RootIsolateToken.instance!;
+  Isolate.spawn(_isolateMain, rootIsolateToken);
+}
+
+Future<void> _isolateMain(RootIsolateToken rootIsolateToken) async {
+  // Register the background isolate with the root isolate.
+  BackgroundIsolateBinaryMessenger.ensureInitialized(rootIsolateToken);
+
+  // You can now use the shared_preferences plugin.
+  SharedPreferences sharedPreferences = await SharedPreferences.getInstance();
+
+  print(sharedPreferences.getBool('isDebug'));
+}
+```
+
+## Limitations of Isolates
+
+If you're coming to Dart from a language with multithreading,
+it's reasonable to expect isolates to behave like threads,
+but that isn't the case.
+Isolates have their own global fields,
+and can only communicate with message passing,
```

Comment:
The nomenclature of this document is "message passing", I wonder if it would be better to use the nomenclature of Dart which is "ports".